docs/topics/scripting/custom-script-deps-tutorial.md

```diff
@@ -0,0 +1,332 @@
+       implementation(project(":script-definition")) // the script definition module
+   }
+   ```
+
+   </tab>
+   <tab title="Groovy" group-key="groovy">
+
+   ```groovy
+   dependencies {
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-common'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-jvm'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-jvm-host'
+       implementation project(':script-definition') // the script definition module
+   }
+   ```
+
+   </tab>
+   </tabs>
+
+2. Create the `src/main/kotlin/` directory in the module and add Kotlin source file, for example, `host.kt`.
+
+3. Define the `main` function for the application. The application should run with one argument – the path to the script file –
+   and execute the script. You'll define the script execution in a separate function `evalFile` in the next step. Declare it empty for now.
+
+   `main` can look like this:
+
+   ```kotlin
+    fun main(vararg args: String) {
+        if (args.size != 1) {
+            println("usage: <app> <script file>")
+        } else {
+            val scriptFile = File(args[0])
+            println("Executing script $scriptFile")
+            evalFile(scriptFile)
+        }
+    }
+   ```
+
+4. Define the script evaluation function. This is where you'll use the script definition. Obtain it by calling `createJvmCompilationConfigurationFromTemplate`
+   with the script definition class as a type parameter. Then call `BasicJvmScriptingHost().eval` passing it the script code and its
+   compilation configuration. `eval` returns an instance of `ResultWithDiagnostics`, so set your function's return type to it.
+
+   ```kotlin
+    fun evalFile(scriptFile: File): ResultWithDiagnostics<EvaluationResult> {
+        val compilationConfiguration = createJvmCompilationConfigurationFromTemplate<ScriptWithMavenDeps>()
+        return BasicJvmScriptingHost().eval(scriptFile.toScriptSource(), compilationConfiguration, null)
+    }
+   ```
+
+5. Adjust the `main` function to print information about the script execution:
+
+   ```kotlin
+    fun main(vararg args: String) {
+        if (args.size != 1) {
+            println("usage: <app> <script file>")
+        } else {
+            val scriptFile = File(args[0])
+            println("Executing script $scriptFile")
+            val res = evalFile(scriptFile)
+            res.reports.forEach {
+                if (it.severity > ScriptDiagnostic.Severity.DEBUG) {
+                    println(" : ${it.message}" + if (it.exception == null) "" else ": ${it.exception}")
+                }
+            }
+        }
+    }
+   ```
+
+You can find the full code [here](https://github.com/Kotlin/kotlin-script-examples/blob/master/jvm/basic/jvm-maven-deps/host/src/main/kotlin/org/jetbrains/kotlin/script/examples/jvm/resolve/maven/host/host.kt)
+
+## Run scripts
+
+To check how your scripting host works, prepare a script to execute and a run configuration.
```

Comment:
I like this intro sentence.
Maybe we should do the same for the previous steps? You have the ending "The next step is creating the scripting host – the component that handles the script execution." and you could easily convert it to the intro sentence of the next section.

Reply:
Added